File: src/main/java/kr/hailor/hailor/entity/Reservation.java
package kr.hailor.hailor.entity;

import jakarta.persistence.*;
import lombok.*;
import org.springframework.format.annotation.DateTimeFormat;

import java.time.LocalDateTime;

@Entity
@Table(name = "reservations")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Reservation {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;

  // 예약자 (User 엔티티와 연관; 예약 신청 시 JWT 인증된 사용자의 정보가 연결됨)
  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name = "user_id")
  private User user;

  // 예약한 디자이너
  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name = "designer_id")
  private Designer designer;

  // 대면/비대면 선택 옵션 (enum 값)
  @Enumerated(EnumType.STRING)
  private ConsultationType type;

  // 예약 날짜 및 시간
  @DateTimeFormat(pattern = "yyyy-MM-dd HH:mm")
  private LocalDateTime reservationDateTime;

  // 예약 시 확정된 가격 (디자이너의 offlinePrice 또는 onlinePrice 중 하나)
  private int price;

  // 예약 취소 여부 (추후 예약 취소 기능 구현 시 사용; 현재 예약 신청에는 필요 없으나 남겨둠)
  private boolean canceled;
}

File: src/main/java/kr/hailor/hailor/enity/Reservation.kt
package kr.hailor.hailor.enity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.ManyToOne
import jakarta.persistence.SequenceGenerator
import jakarta.persistence.Table
import java.time.LocalDate

@Entity
@Table(name = "reservation")
@SequenceGenerator(name = "RESERVATION_SEQ_GENERATOR", sequenceName = "RESERVATION_SEQ", initialValue = 1, allocationSize = 1)
class Reservation(
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "RESERVATION_SEQ_GENERATOR")
    @Column(updatable = false, nullable = false)
    val id: Long = 0,
    @ManyToOne
    val designer: Designer,
    @ManyToOne
    val user: User,
    @Column(nullable = false)
    val reservationDate: LocalDate,
    @Column(nullable = false)
    val slot: Int,
    @Enumerated(EnumType.STRING)
    @Column(length = 20, nullable = false)
    val meetingType: MeetingType,
    @Enumerated(EnumType.STRING)
    @Column(length = 20, nullable = false)
    var status: ReservationStatus,
    @Enumerated(EnumType.STRING)
    @Column(length = 20, nullable = false)
    var paymentMethod: PaymentMethod,
    @Column(length = 100)
    var googleMeetLink: String? = null,
    @Column(length = 50)
    var googleCalendarEventId: String? = null,
    @Column(nullable = false, updatable = false)
    val price: Int,
    @Column(length = 100)
    var paymentId: String? = null,
) : BaseModifiableEntity()

enum class ReservationStatus(
    val description: String,
) {
    RESERVED("예약됨"),
    CONFIRMED("확정됨"),
    FINISHED("종료됨"),
    CANCELED("취소됨"),
    NEED_REFUND("환불 요청됨"),
    REFUNDED("환불됨"),
}

enum class PaymentMethod {
    KAKAO_PAY,
    BANK_TRANSFER,
}
